src/NuGet.Server/Infrastructure/ServerPackageRepository.cs

```diff
@@ -458,17 +458,25 @@ private HashSet<ServerPackage> ReadPackagesFromDisk()
 
                 Parallel.ForEach(packages, package =>
                 {
-                    // Create server package
-                    var serverPackage = CreateServerPackage(package, enableDelisting);
-
-                    // Add the package to the cache, it should not exist already
-                    if (cachedPackages.Contains(serverPackage))
+                    try
                     {
-                        _logger.Log(LogLevel.Warning, "Duplicate package found - {0} {1}", package.Id, package.Version);
+                        // Create server package
+                        var serverPackage = CreateServerPackage(package, enableDelisting);
+
+                        // Add the package to the cache, it should not exist already
```

Comment:
I wrapped the cache code because is the CreateServerPackage call failed we wouldn't have a package to check the cache for. I could definitely see the value of catching the issue in CreateServerPackage but then we'd either need to return a more complex type or use an out param. Alternatively I guess I could create a "TryCreateServerPackage" that follows a similar pattern to methods like Int32.TryParse. It's up to you, happy to address the issue either way.